vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+- `payload` provides an arbitrary byte array containing the contents of the message. VMs define their own message types to include in the `payload`
+
+
+BitSetSignature:
+```
++-----------+----------+---------------------------+
+|   type_id :   uint32 |                   4 bytes |
++-----------+----------+---------------------------+
+|   signers :   []byte |          4 + len(signers) |
++-----------+----------+---------------------------+
+| signature : [96]byte |                  96 bytes |
++-----------+----------+---------------------------+
+                       | 104 + size(signers) bytes |
+                       +---------------------------+
+```
+
+- `typeID` is the ID of this signature type, which is `0x00000000`
+- `signers` encodes a bitset of which validators' signatures are included (a bitset is a byte array where each bit indicates membership of the element at that index in the set)
+- `signature` is an aggregated BLS Multi-Signature of the Unsigned Message
+
+BitSetSignatures are verified within the context of a specific P-Chain height. At a specific P-Chain height, the PlatformVM serves a canonically ordered validator set for the source subnet (validator set is ordered lexicographically by the BLS public key bytes). The signers bitset encodes which validators' signature was included. A value of 1 at index i in `signers` claims that a signature from the validator at index i in the canonical validator set is included in the aggregate signature.
+
+The bitset tells the verifier which BLS public keys should be aggregated to verify the warp message.
+
+Signed Message:
+```
++------------------+------------------+-------------------------------------------------+
+| unsigned_message :  UnsignedMessage |                          size(unsigned_message) |
++------------------+------------------+-------------------------------------------------+
+|        signature :        Signature |                                 size(signature) |
++------------------+------------------+-------------------------------------------------+
+                                      |  size(unsigned_message) + size(signature) bytes |
+                                      +-------------------------------------------------+
+```
+
+## Sending an Avalanche Warp Message
+
+A Blockchain on Avalanche sends an Avalanche Warp Message by coming to agreement on the message that every validator should be willing to sign. As an example, the VM of a Blockchain may define that once a block is accepted, the VM should be willing to sign a message including the block hash in the payload to attest to any other Subnet that the block was accepted. The contents of the payload, how to aggregate the signature (VM-to-VM communication, off-chain relayer, etc.), is left to the VM.
```

Comment:
I don't have a preference but we should be consistent about capitalizing blockchain or Blockchain. I think we use lowercase in earlier parts of the doc